### PR TITLE
[DO NOT MERGE] Native link POC

### PR DESCRIPTION
**Pull request by danielamorse**
## Jira

http://vjira2:8080/browse/BDS-2123

## Summary

POC for a native link component which is not a web component.

## Details

We'd like to provide downstream users with an option to renders a subset of components without Web Component capabilities. This PR shows how we might do that by setting a `noWC` flag.

**Open questions:**
- If we like this approach, how do we want to document it in PL? It creates a duplicate version of every component variation.

## How to test

- Review files changed.
- Checkout code locally and set `_link-replacement.twig` as the default template in `link.twig`. This is the easiest way to test every variation of link.

### CODE REVIEW

File: packages/components/bolt-link/src/_link-replacement.twig [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    class="{{ classes|join(' ') }}"
+    {{ attributes|without("url")|without("href")|without("target")|without("class") }}
+  >
+    {{ macros.slotted_icon(icon, icon_position, "before", true) }}
+    <span class="{{ "c-bolt-link__text" }}">
+      {{- text|default(label)|default("Learn More") -}}
+    </span>
+    {{ macros.slotted_icon(icon, icon_position, "after", true) }}
+  </{{ tagName }}>
+</span>{% endspaceless %}
```
Comment (colbytcook) [inline packages/components/bolt-link/src/_link-replacement.twig:61]: Is this an extra "</span>", I do not see the opening tag

File: packages/components/bolt-link/src/_link-deprecated.twig [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    {% set inner_classes = inner_classes|merge([class]) %}
+  {% elseif class starts with "c-bolt-" == false %}
+    {% set outer_classes = outer_classes|merge([class]) %}
+  {% endif %}
+{% endfor %}
+
+{# Filter out attributes assigned above #}
+{% set filtered_attributes = attributes | without("url") | without("href") | without("target") | without("class") %}
+
+{% set outerTagname = noWC ? "span" : "bolt-link" %}
```
[the remark this answers is not in the document]
Comment (danielamorse) [reply packages/components/bolt-link/src/_link-deprecated.twig]: Somehow you were able to comment on an outdated version of this file. I removed this in my last commit.

File: packages/components/bolt-link/src/_link-deprecated.twig [hunk 3]
```diff
@@ -0,0 +1,110 @@
+{% import "@bolt-components-link/_link-macros.twig" as macros %}
```
[the remark this answers is not in the document]
Comment (danielamorse) [reply packages/components/bolt-link/src/_link-deprecated.twig:1]: Same as https://github.com/boltdesignsystem/bolt/pull/1881#discussion_r434587111

File: packages/components/bolt-link/src/link.twig [hunk 4]
```diff
@@ -1,108 +1,10 @@
...
-    <ssr-keep for="bolt-link" class="{{ "#{base_class}__text" }}">
-      {{- text | default(label) | default("Learn More") -}}
-    </ssr-keep>
-    {{ macros.slotted_icon(icon, icon_position, "after") }}
-  </{{ tagName }}>
-</bolt-link>{% endspaceless %}
+{% if noWC %}
+  {% include "@bolt-components-link/_link-replacement.twig" %}
+{% else %}
+  {% include "@bolt-components-link/_link-deprecated.twig" %}
```
[the remark this answers is not in the document]
Comment (danielamorse) [reply packages/components/bolt-link/src/link.twig:9]: I'm open to that. Let's decide on a convention before we roll this out to other components.

File: packages/components/bolt-link/src/_link-replacement.twig [hunk 5]
```diff
@@ -0,0 +1,61 @@
+{% import "@bolt-components-link/_link-macros.twig" as macros %}
```
[the remark this answers is not in the document]
Comment (danielamorse) [reply packages/components/bolt-link/src/_link-replacement.twig:1]: I thought about this and decided not to. I want to make `_link-deprecated.twig` (or whatever we name it) easy to delete when we move to the new system, so some of the duplication is intentional. Part of the exercise is to figure out what the Twig template looks like after we remove the support for WC (what changes and what can be removed), and keeping them clearly separated helps do that.

**Pull request closed without merging**